packages/shared/src/authz/fragments/workspaceSso.spec.ts

```diff
@@ -12,6 +14,39 @@ import {
 import { just, nothing } from 'true-myth/maybe'
 
 describe('maybeMemberRoleWithValidSsoSessionIfNeeded returns a function, that', () => {
+  it.each([new WorkspaceNoAccessError(), new WorkspaceNotFoundError()])(
+    'remaps workspace loader error $code into WorkspaceNoAccessError',
+    async (expectedError) => {
+      const result = maybeMemberRoleWithValidSsoSessionIfNeeded({
+        getWorkspace: async () => err(expectedError),
+        getWorkspaceRole: async () => err(new WorkspaceRoleNotFoundError()),
+        getWorkspaceSsoProvider: async () =>
+          err(new WorkspaceSsoProviderNotFoundError()),
+        getWorkspaceSsoSession: async () => err(new WorkspaceSsoSessionNotFoundError())
+      })({
+        userId: cryptoRandomString({ length: 10 }),
+        workspaceId: cryptoRandomString({ length: 10 })
+      })
+      await expect(result).resolves.toStrictEqual(
+        just(err(new WorkspaceNoAccessError()))
+      )
+    }
+  )
+  it('throws Uncovered error for unknown loader errors', async () => {
+    await expect(
+      maybeMemberRoleWithValidSsoSessionIfNeeded({
+        // @ts-expect-error testing the unexpected error case here
```

Comment:
if you put a @ts-expect-error here, its not gonna show errors even if there are completely missing dependencies in the factory, which is not good

maybe if you put it lower, inside of the actual body of the getWorkspace implementation, then it won't apply to the entire factory. if not, maybe its better to do `error as AcceptableErrorType`